operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/IDMapper.java

```diff
@@ -0,0 +1,18 @@
+package io.javaoperatorsdk.operator.processing.event.source;
+
+import java.util.function.Function;
+
+public interface IDMapper<R> extends Function<R, String> {
```

Comment:
I'd rather have an explicit interface with an actual method which name we control rather than a Function extension. Easier to extend at a later time and more self-documenting than a random `apply` method in implementors, imo.
I'm also unsure about the purpose of this, actually. :(

Reply:
yep, will make it more explicit, not extending the function. Tried to explain here, why it is needed:

https://github.com/java-operator-sdk/java-operator-sdk/blob/44987c93543814e7c3b8d5f1246adb7a8ab47f5e/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/ExternalResourceCachingEventSource.java#L15-L32

Reply:
pushed. 

Can try to explain more in depth here in the comment why this needed. But basically we need to have the information if it's a new resource is a new version or just the same as an other. With key check if it referring to same object with equals we check if it changed or not.